lib/util/semaphore.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2018 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package util
```

Comment:
(We should split `package util` into `structutil`, `netutil`, `stringsutil`, `syncutil` to regain some semblance of order instead of random-bag-of-crap... but not in this PR...)

Reply:
We already have a few of these. Actually, a semaphore might also belong to `lib/sync`? Given I am moving it around in this PR, we might as well move it to an appropriate location now.